src/main/java/hudson/plugins/spotinst/api/SpotinstApi.java

```diff
@@ -313,6 +317,75 @@ public static Boolean azureVmDetach(String groupId, String vmId, String accountI
     }
     //endregion
 
+    //Redis
+    //TODO Liron - change name to getLock
+    public static <T> T getRedisValue(String groupId, String accountId) throws ApiException {
+        T retVal      = null;
+
+        Map<String, String>    headers     = buildHeaders();
+
+        Map<String, String>    queryParams = buildQueryParams(accountId);
+
+        RestResponse response =
+                RestClient.sendGet(SPOTINST_API_HOST + "/aws/ec2/group/" + groupId + "/jenkinsPlugin", headers, queryParams);
+
+        RedisGetValueResponse<T> redisValue = getCastedResponse(response, RedisGetValueResponse.class);
+
+        if (redisValue.getResponse().getItems().size() > 0) {
+            retVal = redisValue.getResponse().getItems().get(0);
+        }
+
+        return retVal;
+    }
+
+    //TODO Liron - change method name lock
+    public static String setRedisKey(String lockKey, String accountId, String lockValue, Integer ttl) throws ApiException {
+        String retVal      = null;
+
+        Map<String, String>    headers     = buildHeaders();
+
+        Map<String, String>    queryParams = buildQueryParams(accountId);
+
+        RedisSetKeyRequest request = new RedisSetKeyRequest();
+        request.setGroupId(lockKey);
+        request.setControllerIdentifier(lockValue);
+        request.setTtl(ttl);
+
+        String body = JsonMapper.toJson(request);
+
+        RestResponse response =
+                RestClient.sendPost(SPOTINST_API_HOST + "/aws/ec2/group/jenkinsPlugin", body, headers, queryParams);
+
+        RedisSetKeyResponse redisValue = getCastedResponse(response, RedisSetKeyResponse.class);
+
+        if (redisValue.getResponse().getItems().size() > 0) {
+            retVal = redisValue.getResponse().getItems().get(0);
+        }
+
+        return retVal;
+    }
+
+    //TODO Liron - change name to unlock
+    public static Integer deleteRedisKey(String groupId, String accountId) throws ApiException {
+        Integer retVal      = null;
+
+        Map<String, String>    headers     = buildHeaders();
+
+        Map<String, String>    queryParams = buildQueryParams(accountId);
+
+        RestResponse response =
+                RestClient.sendDelete(SPOTINST_API_HOST + "/aws/ec2/group/" + groupId + "/jenkinsPlugin", headers, queryParams);
```

Comment:
Add the suffix "lock" to the url (and change the optimizer)

Reply:
done